### PR TITLE
(master) Update to 2018 edition

### PR DESCRIPTION
**Pull request by longwusha**
## Description
Update to 2018 edition for master branch

## Related Issue
#54 

## How This Has Been Tested
- [x] `cargo run --example kaleidoscope --no-default-features --features "llvm7-0"`
- [x] `cargo test --no-default-features --features "llvm7-0"`

**Other branch need more work.**

## Option\<Breaking Changes\>
minimum rust version change to 1.31

## Checklist

- [x] I have read the [Contributing Guide](https://github.com/TheDan64/inkwell/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by longwusha** [comment]
@TheDan64  I reimplemented this PR with minimal changes (without modifying the code format). Most of the CI tests have now passed. Some errors have occurred in the CI tests of llvm 3.8, llvm 5.0, and llvm 6.0, but the error log does not indicate what the problem is. I think I need your help.

**Comment by TheDan64** [comment]
LLVM sometimes randomly segfaults on travis which are the failures you're seeing. I haven't been able to reproduce it locally so I don't think it's specific to anything inkwell is doing. Anyhow I've restarted those jobs

File: src/builder.rs [hunk 1]
```diff
@@ -667,6 +667,11 @@ impl Builder {
     /// if the leftmost bit was one.
     ///
     /// ```rust,no_run
+	/// //fix doc error about overflowing_literals
+	/// //rendered rfc: https://github.com/rust-lang/rfcs/blob/master/text/2438-deny-integer-literal-overflow-lint.md
+	/// //tracking issue: https://github.com/rust-lang/rust/issues/54502
```
Comment (TheDan64) [inline src/builder.rs]: This should be prefixed by 4 spaces, not a tab

Comment (longwusha) [reply src/builder.rs]: I have fixed this problem, but the CI test crashes again, please restart the test tasks again. Also, is there any way I can restart these test tasks myself?

**Comment by TheDan64** [comment]
Thanks!

**Pull request merged**